subworkflows/local/utils_nfcore_mag_pipeline/main.nf

```diff
@@ -89,13 +89,36 @@ workflow PIPELINE_INITIALISATION {
         }
     }
 
+    // Check that all short read sequencing platforms are the same within a group
+    ch_raw_short_reads
+        .map { meta, sr -> [ meta.group, meta.sr_platform ] }
```

Comment:
What happens here if `meta.group` is not specified? AFAIK it's an optional column and if you're not using `--coassemble_group` then downstream, `meta.group` is set to `meta.id`.